httpsredirect.go

```diff
@@ -1,27 +1,19 @@
 package gokrazy
 
 import (
+	"net"
 	"net/http"
-	"strings"
 )
 
-// TODO: Configurable https-port
-func httpsRedirect(handler http.Handler) func(w http.ResponseWriter, r *http.Request) {
-	return func(w http.ResponseWriter, r *http.Request) {
-		// get hostname without port
-		target := r.URL.Host
-		if port := ":" + r.URL.Port(); strings.HasSuffix(target, port) {
-			target = strings.TrimSuffix(target, port)
-		}
-
-		// serve localhost on http
-		if target == "localhost" {
-			handler.ServeHTTP(w, r)
-			return
-		}
-
-		r.URL.Host = r.Host
-		r.URL.Scheme = "https"
-		http.Redirect(w, r, r.URL.String(), http.StatusFound) // Redirect to https
+func httpsRedirect(w http.ResponseWriter, r *http.Request) {
+	// get hostname without port
+	ip := net.ParseIP(r.RemoteAddr)
+	if IsInPrivateNet(ip) {
```

Comment:
This is a bit too loose: we need a variant that not only checks the IP is private, but specifically that it’s IPv4/IPv6 localhost.

Reply:
Could you grab this? Not sure what to do, sorry.

Reply:
What IP do you connect to?
If you connect to localhost I think this will work: https://golang.org/src/net/ip.go?s=3355:3385#L111

But if you connect to "the IP of eth0", like 192.168.10.10, then it's more difficult.

Reply:
Lets support this only for loopback, localhost is fine for the purpose.